sql/parser/builtins.go

```diff
@@ -311,10 +312,19 @@ var builtins = map[string][]builtin{
 		builtin{
 			types:      typeList{},
 			returnType: DummyFloat,
+			impure:     true,
 			fn: func(args DTuple) (Datum, error) {
 				return DFloat(rand.Float64()), nil
 			},
 		},
+		builtin{
+			types:      typeList{intType},
+			returnType: DummyFloat,
+			impure:     true,
+			fn: func(args DTuple) (Datum, error) {
+				return DFloat(rand.New(rand.NewSource(int64(args[0].(DInt)))).Float64()), nil
```

Comment:
Yeah, the implementation was busted and it doesn't seem easy to fix with our current eval infrastructure.